src/components/HomePage/SearchResultsArea/Offer/OfferContent.js

```diff
@@ -1,22 +1,89 @@
+import OfferContentListItem from "./OfferContentListItem";
 
 import LoadingMagnifyGlass from "./loading_magnify_glass_svg";
 import { useDesktop } from "../../../../utils/media-queries";
+import useSession from "../../../../hooks/useSession";
 
 import useSearchResultsWidgetStyles from "../SearchResultsWidget/searchResultsWidgetStyles";
 import LOADING_MESSAGES from "./offerLoadingMessages";
-import { DateRange, LocationCity } from "@material-ui/icons";
-import { format, parseISO } from "date-fns";
+import {
+    DateRange, LocationOn, Work, Schedule, MonetizationOn, MoneyOff,
+    FindInPage, Info,
+} from "@material-ui/icons";
+import { format, parseISO, formatDistanceToNowStrict } from "date-fns";
+import ChipList from "./ChipList";
+import AdminReasonModal from "./AdminReasonModal";
+
+import JOB_OPTIONS from "../../SearchArea/AdvancedSearch/JobOptions";
+
+import { addSnackbar } from "../../../../actions/notificationActions";
+
+import { connect } from "react-redux";
+import { getHumanError } from "../../../../utils/offer/OfferUtils";
+import { Skeleton } from "@material-ui/lab";
+import OfferDescription from "./OfferDescription";
+import OfferVisibilityOptions from "./OfferVisibilityOptions";
 
 const getRandomOngoingSearchMessage = () => LOADING_MESSAGES[Math.floor(Math.random() * LOADING_MESSAGES.length)];
 
-const OfferContent = ({ offer, loading }) => {
-    const classes = useSearchResultsWidgetStyles({ isMobile: !useDesktop() });
-    if (loading) {
+const defaultLogo = require("./default_icon.svg");
+
+const OfferContent = ({ addSnackbar, offer, loading, isPage }) => {
+
+    const { data, isValidating, error, isLoggedIn } = useSession();
+    const sessionData = (!isValidating && !error && isLoggedIn) ? data : null;
+    const isMobile = !useDesktop();
+    const classes = useSearchResultsWidgetStyles({ isMobile, isPage, loading });
+
+    const [visibilityState, setVisibilityState] = useState({
+        isVisible: undefined,
+        isDisabled: undefined,
+        isBlocked: undefined,
+    });
+    visibilityState.isDisabled = offer?.isHidden && offer?.hiddenReason === "ADMIN_REQUEST";
+    visibilityState.isVisible = !offer?.isHidden;
+    visibilityState.isBlocked = offer?.isHidden && offer?.hiddenReason === "COMPANY_BLOCKED";
+
+    const [visibilityError, setVisibilityError] = useState(null);
+    const [showAdminReasonModal, setShowAdminReasonModal] = useState(false);
+
+    const dealWithPromiseError = (err) => {
+        if (Array.isArray(err) && err.length > 0) {
+            if (Object.prototype.hasOwnProperty.call(err[0], "msg"))
+                setVisibilityError(getHumanError(err[0]?.msg));
+            else
+                setVisibilityError(getHumanError(err[0]));
+        }
+        addSnackbar({
+            message: visibilityError ? visibilityError : "Unexpected Error. Please try again later.",
+            key: visibilityError,
+        });
+    };
+
+    const getDisabledOfferMessage = () => (
```

Comment:
Wrap these functions in `useCallback`, so as to avoid creating a new function every render.

It works like useEffect, you give it the function (what you currently have) as the first arg, and the second is an array of dependences. Whenever the dependencies change, a new function is created. In this case the dependencies should be `sessionData?.isAdmin` and `offer.adminReason`, for example